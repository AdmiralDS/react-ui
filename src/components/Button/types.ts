export type Appearance = 'primary' | 'secondary' | 'ghost' | 'white' | 'danger' | 'success';
export type Dimension = 'xl' | 'l' | 'm' | 's';

export interface StyledButtonProps {
  appearance?: Appearance;
  dimension?: Dimension;
  displayAsDisabled?: boolean;
  $loading?: boolean;
}
