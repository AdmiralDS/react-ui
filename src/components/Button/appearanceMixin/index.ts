import { css } from 'styled-components';
import { primaryAppearanceMixin } from './primaryAppearanceMixin';
import { secondaryAppearanceMixin } from './secondaryAppearanceMixin';
import { tertiaryAppearanceMixin } from './tertiaryAppearanceMixin';
import { ghostAppearanceMixin } from './ghostAppearanceMixin';
import { whiteAppearanceMixin } from './whiteAppearanceMixin';
import { dangerAppearanceMixin } from './dangerAppearanceMixin';
import { successAppearanceMixin } from './successAppearanceMixin';
import { focusVisibleStyle } from '#src/components/Button/appearanceMixin/focusVisibleStyle';

export const appearanceMixin = css`
  &[data-appearance~='primary']:not(.button-group > button) {
    ${primaryAppearanceMixin}
  }
  &[data-appearance~='secondary']:not(.button-group > button) {
    ${secondaryAppearanceMixin}
  }
  &[data-appearance~='tertiary']:not(.button-group > button) {
    ${tertiaryAppearanceMixin}
  }
  &[data-appearance~='ghost']:not(.button-group > button) {
    ${ghostAppearanceMixin}
  }
  &[data-appearance~='white']:not(.button-group > button) {
    ${whiteAppearanceMixin}
  }
  &[data-appearance~='danger']:not(.button-group > button) {
    ${dangerAppearanceMixin}
  }
  &[data-appearance~='success']:not(.button-group > button) {
    ${successAppearanceMixin}
  }
  &:is(.button-group[data-appearance='primary'] > button) {
    ${primaryAppearanceMixin}
  }
  &:is(.button-group[data-appearance='secondary'] > button) {
    ${secondaryAppearanceMixin}
  }
  &:is(.button-group[data-appearance='tertiary'] > button) {
    ${tertiaryAppearanceMixin}
  }

  ${focusVisibleStyle}
`;
