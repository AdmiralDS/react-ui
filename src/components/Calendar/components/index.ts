export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
