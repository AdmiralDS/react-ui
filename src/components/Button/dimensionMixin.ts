import styled, { css } from 'styled-components';
import { typography } from '#src/components/Typography';
import type { Dimension } from '#src/components/Button/types';

export const ButtonIconContainer = styled.div``;

function getHorizontalIconPadding(dimension: Dimension) {
  switch (dimension) {
    case 's':
      return 13;
    case 'm':
      return 17;
    case 'l':
      return 21;
    case 'xl':
    default:
      return 29;
  }
}

/**
 * @deprecated Помечено как deprecated в версии 5.2.0, будет удалено в 9.x.x версии.
 * Взамен используйте iconStart или iconEnd
 *
 * css mixin для установки правильного отступа у иконки
 **/
export const PaddingForButtonWithIconLeft = css<{ $displayAsSquare?: boolean }>`
  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${(props) => !props.$displayAsSquare && `padding-left: ${getHorizontalIconPadding('xl')}px;`}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    ${(props) => !props.$displayAsSquare && `padding-left: ${getHorizontalIconPadding('l')}px;`}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    ${(props) => !props.$displayAsSquare && `padding-left: ${getHorizontalIconPadding('m')}px;`}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    ${(props) => !props.$displayAsSquare && `padding-left: ${getHorizontalIconPadding('s')}px;`}
  }
`;

/**
 * @deprecated Помечено как deprecated в версии 5.2.0, будет удалено в 9.x.x версии.
 * Взамен используйте iconStart или iconEnd
 *
 * css mixin для установки правильного отступа у иконки
 **/
export const PaddingForButtonWithIconRight = css<{ $displayAsSquare?: boolean }>`
  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${(props) => !props.$displayAsSquare && `padding-right: ${getHorizontalIconPadding('xl')}px;`}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    ${(props) => !props.$displayAsSquare && `padding-right: ${getHorizontalIconPadding('l')}px;`}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    ${(props) => !props.$displayAsSquare && `padding-right: ${getHorizontalIconPadding('m')}px;`}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    ${(props) => !props.$displayAsSquare && `padding-right: ${getHorizontalIconPadding('s')}px;`}
  }
`;

const defaultDimensionMixin = css<{ $displayAsSquare?: boolean }>`
  padding: 0;
  height: 56px;
  ${(props) => (props.$displayAsSquare ? 'width: 56px;' : 'padding: 0 29px;')}
  ${ButtonIconContainer} {
    width: 24px;
    height: 24px;
  }

  ${typography['Button/Button 1']}
`;

export const dimensionMixin = css<{ $displayAsSquare?: boolean }>`
  ${defaultDimensionMixin}

  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${defaultDimensionMixin}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    height: 48px;
    ${(props) => (props.$displayAsSquare ? 'width: 48px;' : 'padding: 0 21px;')}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    height: 40px;
    ${(props) => (props.$displayAsSquare ? 'width: 40px;' : 'padding: 0 17px;')}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    height: 32px;
    ${(props) => (props.$displayAsSquare ? 'width: 32px;' : 'padding: 0 13px;')}
    ${ButtonIconContainer} {
      width: 20px;
      height: 20px;
    }
    ${typography['Button/Button 2']}
  }
`;
