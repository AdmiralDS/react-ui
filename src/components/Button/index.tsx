import { Children, forwardRef } from 'react';
import type { ButtonHTMLAttributes, ReactNode } from 'react';
import type { css } from 'styled-components';
import styled from 'styled-components';

import type { ButtonAppearance, Dimension, StyledButtonProps } from './types';
import { SpinnerIcon } from '#src/components/Spinner/SpinnerIcon';
import { appearanceMixin } from './appearanceMixin';
import { ButtonIconContainer, dimensionMixin } from './dimensionMixin';
import { mediumGroupBorderRadius } from '#src/components/themes/borderRadius';
import { skeletonAnimationMixin } from '#src/components/skeleton/animation';

export type { ButtonAppearance } from './types';
export * from './dimensionMixin';

type IconPlace = 'left' | 'right';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /** Внешний вид кнопки */
  appearance?: ButtonAppearance;

  /** Размер кнопки */
  dimension?: Dimension;

  /** Оставаясь активной для нажатия, кнопка отображается в заблокированном стиле */
  displayAsDisabled?: boolean;

  /** Отображать кнопку квадратной*/
  displayAsSquare?: boolean;

  /** Состояние загрузки */
  loading?: boolean;

  /** Состояние скелетона */
  skeleton?: boolean;

  /**
   * @deprecated Помечено как deprecated в версии 5.2.0, будет удалено в 9.x.x версии.
   * Взамен используйте iconStart или iconEnd
   *
   * Иконка кнопки
   */
  icon?: ReactNode;

  /** Иконка перед текстом кнопки */
  iconStart?: ReactNode;

  /** Иконка после текста кнопки */
  iconEnd?: ReactNode;

  /**
   * @deprecated Помечено как deprecated в версии 5.2.0, будет удалено в 9.x.x версии.
   * Взамен используйте iconStart или iconEnd
   *
   * Расположение иконки кнопки
   */
  iconPlace?: IconPlace;

  /** Позволяет добавлять миксин для кнопок, созданный с помощью styled css  */
  buttonCssMixin?: ReturnType<typeof css>;
}

const ButtonContent = styled.div<{ $addPadding: number }>`
  vertical-align: top;

  display: inline-flex;
  gap: 8px;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: center;
  max-width: calc(100% - ${(p) => p.$addPadding}px);

  > * {
    display: inline-block;
    flex: 0 1 auto;
    white-space: nowrap;
  }
  > ${ButtonIconContainer} {
    flex: 0 0 auto;
  }

  height: 24px;
  & > svg {
    width: 24px;
    height: 24px;
  }

  .button-group[data-dimension='m'] &&,
  .button-group[data-dimension='l'] &&,
  .button-group[data-dimension='xl'] && {
    height: 24px;
    & > svg {
      width: 24px;
      height: 24px;
    }
  }
  .button-group[data-dimension='s'] &&,
  [data-dimension='s'] & {
    height: 20px;
    & > svg {
      width: 20px;
      height: 20px;
    }
  }
`;

const SpinnerContainer = styled(ButtonIconContainer)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  container-type: inline-size;
  .button-group[data-appearance='primary'] & {
    & path {
      fill: ${(p) => `var(--admiral-color-Special_StaticWhite, ${p.theme.color['Special/Static White']})`};
    }
  }
  .button-group[data-appearance='secondary'] &,
  .button-group[data-appearance='tertiary'] & {
    & path {
      fill: ${(p) => `var(--admiral-color-Primary_Primary60Main, ${p.theme.color['Primary/Primary 60 Main']})`};
    }
  }
`;

const AdditionalPadding = styled.div`
  display: inline-block;
  width: 2px;
`;

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      appearance = 'primary',
      dimension = 'xl',
      type = 'button',
      loading = false,
      skeleton = false,
      iconStart,
      iconEnd,
      icon,
      iconPlace = 'left',
      children,
      buttonCssMixin,
      displayAsDisabled,
      displayAsSquare,
      ...props
    },
    ref,
  ) => {
    const spinnerInverse = appearance !== 'secondary' && appearance !== 'tertiary' && appearance !== 'ghost';
    const hasIconStart = !!iconStart || (!!icon && iconPlace === 'left');
    const hasIconEnd = !!iconEnd || (!!icon && iconPlace === 'right');
    const additionalPadding = (!displayAsSquare && !hasIconStart ? 2 : 0) + (!displayAsSquare && !hasIconEnd ? 2 : 0);

    return (
      <StyledButton
        ref={ref}
        $appearance={appearance}
        $dimension={dimension}
        type={type}
        $loading={loading}
        $skeleton={skeleton}
        $buttonCssMixin={buttonCssMixin}
        $displayAsSquare={displayAsSquare}
        $displayAsDisabled={displayAsDisabled}
        {...props}
      >
        {loading && (
          <SpinnerContainer>
            <SpinnerIcon $inverse={spinnerInverse} />
          </SpinnerContainer>
        )}
        {!displayAsSquare && !hasIconStart && <AdditionalPadding />}
        <ButtonContent $addPadding={additionalPadding}>
          {hasIconStart ? <ButtonIconContainer>{iconStart || icon}</ButtonIconContainer> : null}
          {Children.toArray(children).map((child, index) =>
            typeof child === 'string' ? <div key={child + index}>{child}</div> : child,
          )}
          {hasIconEnd ? <ButtonIconContainer>{iconEnd || icon}</ButtonIconContainer> : null}
        </ButtonContent>
        {!displayAsSquare && !hasIconEnd && <AdditionalPadding />}
      </StyledButton>
    );
  },
);

const StyledButton = styled.button.attrs<
  StyledButtonProps & {
    'data-dimension'?: Dimension;
    'data-appearance'?: string;
  }
>((props) => ({
  'data-dimension': props.$dimension,
  'data-appearance': [props.$appearance, props.$displayAsDisabled ? 'disabled' : undefined]
    .filter((val) => val !== undefined)
    .join(' '),
}))<StyledButtonProps>`
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  white-space: nowrap;
  border: none;
  border-radius: ${(p) =>
    p.$skeleton ? 0 : `var(--admiral-border-radius-Medium, ${mediumGroupBorderRadius(p.theme.shape)})`};
  appearance: none;
  vertical-align: middle;
  cursor: ${(p) => (p.disabled ? 'not-allowed' : 'pointer')};
  ${(p) => (p.$loading || p.$skeleton) && 'pointer-events: none'};

  ${appearanceMixin};
  ${dimensionMixin};
  ${(p) => p.$buttonCssMixin};
  ${({ $skeleton }) => $skeleton && skeletonAnimationMixin};

  ${ButtonContent} {
    ${(p) => (p.$loading || p.$skeleton ? 'visibility: hidden;' : '')}
  }
`;

Button.displayName = 'Button';
