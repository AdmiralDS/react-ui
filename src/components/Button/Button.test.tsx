import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LIGHT_THEME } from '#src/components/themes';
import { Button } from '#src/components/Button';

const BUTTON_TEXT = 'Button';

describe('Button', () => {
  test('should render component', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render component with text', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button>{BUTTON_TEXT}</Button>
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders the button', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button>{BUTTON_TEXT}</Button>
      </ThemeProvider>,
    );
    expect(screen.getByRole('button')).toHaveTextContent(`${BUTTON_TEXT}`);
  });

  test('should focus on component when user on press key tab', async () => {
    const user = userEvent.setup();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button>{BUTTON_TEXT}</Button>
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    await user.tab();
    expect(screen.getByRole('button')).toHaveFocus();
  });

  test('should dont focus component when user on press key tab', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button disabled>{BUTTON_TEXT}</Button>
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    userEvent.tab();
    expect(document.body).toHaveFocus();
  });

  test('should call onClick when user clicks on component', () => {
    const handleClick = jest.fn();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Button onClick={handleClick}>{BUTTON_TEXT}</Button>
      </ThemeProvider>,
    );
    fireEvent.click(screen.getByText(`${BUTTON_TEXT}`));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
