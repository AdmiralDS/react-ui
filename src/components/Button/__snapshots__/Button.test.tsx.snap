// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  border: none;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  vertical-align: center;
  padding: 0;
  height: 56px;
  padding: 0 30px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0[data-appearance~='primary'] {
  background-color: #0062FF;
  color: #FFFFFF;
  border: 1px solid #0062FF;
}

.c0[data-appearance~='primary'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='primary']:focus,
.c0[data-appearance~='primary']:hover {
  background-color: #0046E2;
  border-color: #0046E2;
}

.c0[data-appearance~='primary']:active {
  background-color: #0132B0;
  border-color: #0132B0;
}

.c0[data-appearance~='primary'][data-appearance~='disabled'],
.c0[data-appearance~='primary']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='primary'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='primary']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='secondary'] {
  background-color: transparent;
  color: #0062FF;
  border: 1px solid #0062FF;
}

.c0[data-appearance~='secondary'] *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary']:focus,
.c0[data-appearance~='secondary']:hover {
  background-color: #0000000D;
  color: #0062FF;
  border-color: #0062FF;
}

.c0[data-appearance~='secondary']:focus *[fill^='#'],
.c0[data-appearance~='secondary']:hover *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary']:active {
  background-color: #0000001F;
  color: #0062FF;
  border-color: #0062FF;
}

.c0[data-appearance~='secondary']:active *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary'][data-appearance~='disabled'],
.c0[data-appearance~='secondary']:disabled {
  background-color: transparent;
  color: #B1B5BB;
  border-color: #B1B5BB;
}

.c0[data-appearance~='secondary'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='secondary']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='ghost'] {
  background-color: transparent;
  color: #0062FF;
  border: 1px solid transparent;
}

.c0[data-appearance~='ghost'] *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost']:focus,
.c0[data-appearance~='ghost']:hover {
  background-color: #0000000D;
  color: #0062FF;
  border-color: transparent;
}

.c0[data-appearance~='ghost']:focus *[fill^='#'],
.c0[data-appearance~='ghost']:hover *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost']:active {
  background-color: #0000001F;
  color: #0062FF;
  border-color: transparent;
}

.c0[data-appearance~='ghost']:active *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost'][data-appearance~='disabled'],
.c0[data-appearance~='ghost']:disabled {
  background-color: transparent;
  color: #B1B5BB;
  border-color: transparent;
}

.c0[data-appearance~='ghost'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='ghost']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='white'] {
  background-color: transparent;
  color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c0[data-appearance~='white'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white']:focus,
.c0[data-appearance~='white']:hover {
  background-color: #FFFFFF0F;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0[data-appearance~='white']:focus *[fill^='#'],
.c0[data-appearance~='white']:hover *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white']:active {
  background-color: #FFFFFF1F;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0[data-appearance~='white']:active *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white'][data-appearance~='disabled'],
.c0[data-appearance~='white']:disabled {
  border-color: #4E5158;
  color: #4E5158;
}

.c0[data-appearance~='white'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='white']:disabled *[fill^='#'] {
  fill: #4E5158;
}

.c0[data-appearance~='danger'] {
  background-color: #D92020;
  color: #FFFFFF;
  border: 1px solid #D92020;
}

.c0[data-appearance~='danger'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='danger']:focus,
.c0[data-appearance~='danger']:hover {
  background-color: #BA1717;
  border-color: #BA1717;
}

.c0[data-appearance~='danger']:active {
  background-color: #981111;
  border-color: #981111;
}

.c0[data-appearance~='danger'][data-appearance~='disabled'],
.c0[data-appearance~='danger']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='danger'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='danger']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='success'] {
  background-color: #1BA049;
  color: #FFFFFF;
  border: 1px solid #1BA049;
}

.c0[data-appearance~='success'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='success']:focus,
.c0[data-appearance~='success']:hover {
  background-color: #128238;
  border-color: #128238;
}

.c0[data-appearance~='success']:active {
  background-color: #0B682A;
  border-color: #0B682A;
}

.c0[data-appearance~='success'][data-appearance~='disabled'],
.c0[data-appearance~='success']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='success'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='success']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-dimension='xl'] {
  padding: 0;
  height: 56px;
  padding: 0 30px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0[data-dimension='l'] {
  height: 48px;
  padding: 0 22px;
}

.c0[data-dimension='m'] {
  height: 40px;
  padding: 0 18px;
}

.c0[data-dimension='s'] {
  height: 32px;
  padding: 0 14px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0:hover {
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c2 {
  vertical-align: top;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  overflow: hidden;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
}

.c2 > * {
  display: inline-block;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  white-space: nowrap;
}

.c2 > *:not(:first-child) {
  margin-left: 8px;
}

.c2 > svg {
  width: 24px;
  height: 24px;
}

[data-dimension='s'] .c1 {
  height: 20px;
}

[data-dimension='s'] .c1 > svg {
  width: 20px;
  height: 20px;
}

.c2 span {
  visibility: visible;
}

<div>
  <button
    class="c0"
    data-appearance="primary"
    data-dimension="xl"
    type="button"
  >
    <span
      class="c1 c2"
    />
  </button>
</div>
`;

exports[`Button should render component with text 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  border: none;
  border-radius: 4px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  vertical-align: center;
  padding: 0;
  height: 56px;
  padding: 0 30px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0[data-appearance~='primary'] {
  background-color: #0062FF;
  color: #FFFFFF;
  border: 1px solid #0062FF;
}

.c0[data-appearance~='primary'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='primary']:focus,
.c0[data-appearance~='primary']:hover {
  background-color: #0046E2;
  border-color: #0046E2;
}

.c0[data-appearance~='primary']:active {
  background-color: #0132B0;
  border-color: #0132B0;
}

.c0[data-appearance~='primary'][data-appearance~='disabled'],
.c0[data-appearance~='primary']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='primary'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='primary']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='secondary'] {
  background-color: transparent;
  color: #0062FF;
  border: 1px solid #0062FF;
}

.c0[data-appearance~='secondary'] *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary']:focus,
.c0[data-appearance~='secondary']:hover {
  background-color: #0000000D;
  color: #0062FF;
  border-color: #0062FF;
}

.c0[data-appearance~='secondary']:focus *[fill^='#'],
.c0[data-appearance~='secondary']:hover *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary']:active {
  background-color: #0000001F;
  color: #0062FF;
  border-color: #0062FF;
}

.c0[data-appearance~='secondary']:active *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='secondary'][data-appearance~='disabled'],
.c0[data-appearance~='secondary']:disabled {
  background-color: transparent;
  color: #B1B5BB;
  border-color: #B1B5BB;
}

.c0[data-appearance~='secondary'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='secondary']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='ghost'] {
  background-color: transparent;
  color: #0062FF;
  border: 1px solid transparent;
}

.c0[data-appearance~='ghost'] *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost']:focus,
.c0[data-appearance~='ghost']:hover {
  background-color: #0000000D;
  color: #0062FF;
  border-color: transparent;
}

.c0[data-appearance~='ghost']:focus *[fill^='#'],
.c0[data-appearance~='ghost']:hover *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost']:active {
  background-color: #0000001F;
  color: #0062FF;
  border-color: transparent;
}

.c0[data-appearance~='ghost']:active *[fill^='#'] {
  fill: #0062FF;
}

.c0[data-appearance~='ghost'][data-appearance~='disabled'],
.c0[data-appearance~='ghost']:disabled {
  background-color: transparent;
  color: #B1B5BB;
  border-color: transparent;
}

.c0[data-appearance~='ghost'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='ghost']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='white'] {
  background-color: transparent;
  color: #FFFFFF;
  border: 1px solid #FFFFFF;
}

.c0[data-appearance~='white'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white']:focus,
.c0[data-appearance~='white']:hover {
  background-color: #FFFFFF0F;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0[data-appearance~='white']:focus *[fill^='#'],
.c0[data-appearance~='white']:hover *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white']:active {
  background-color: #FFFFFF1F;
  color: #FFFFFF;
  border-color: #FFFFFF;
}

.c0[data-appearance~='white']:active *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='white'][data-appearance~='disabled'],
.c0[data-appearance~='white']:disabled {
  border-color: #4E5158;
  color: #4E5158;
}

.c0[data-appearance~='white'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='white']:disabled *[fill^='#'] {
  fill: #4E5158;
}

.c0[data-appearance~='danger'] {
  background-color: #D92020;
  color: #FFFFFF;
  border: 1px solid #D92020;
}

.c0[data-appearance~='danger'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='danger']:focus,
.c0[data-appearance~='danger']:hover {
  background-color: #BA1717;
  border-color: #BA1717;
}

.c0[data-appearance~='danger']:active {
  background-color: #981111;
  border-color: #981111;
}

.c0[data-appearance~='danger'][data-appearance~='disabled'],
.c0[data-appearance~='danger']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='danger'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='danger']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-appearance~='success'] {
  background-color: #1BA049;
  color: #FFFFFF;
  border: 1px solid #1BA049;
}

.c0[data-appearance~='success'] *[fill^='#'] {
  fill: #FFFFFF;
}

.c0[data-appearance~='success']:focus,
.c0[data-appearance~='success']:hover {
  background-color: #128238;
  border-color: #128238;
}

.c0[data-appearance~='success']:active {
  background-color: #0B682A;
  border-color: #0B682A;
}

.c0[data-appearance~='success'][data-appearance~='disabled'],
.c0[data-appearance~='success']:disabled {
  background-color: #E5E7EB;
  color: #B1B5BB;
  border-color: #E5E7EB;
}

.c0[data-appearance~='success'][data-appearance~='disabled'] *[fill^='#'],
.c0[data-appearance~='success']:disabled *[fill^='#'] {
  fill: #B1B5BB;
}

.c0[data-dimension='xl'] {
  padding: 0;
  height: 56px;
  padding: 0 30px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0[data-dimension='l'] {
  height: 48px;
  padding: 0 22px;
}

.c0[data-dimension='m'] {
  height: 40px;
  padding: 0 18px;
}

.c0[data-dimension='s'] {
  height: 32px;
  padding: 0 14px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c0:hover {
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c2 {
  vertical-align: top;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  overflow: hidden;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
}

.c2 > * {
  display: inline-block;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  white-space: nowrap;
}

.c2 > *:not(:first-child) {
  margin-left: 8px;
}

.c2 > svg {
  width: 24px;
  height: 24px;
}

[data-dimension='s'] .c1 {
  height: 20px;
}

[data-dimension='s'] .c1 > svg {
  width: 20px;
  height: 20px;
}

.c2 span {
  visibility: visible;
}

<div>
  <button
    class="c0"
    data-appearance="primary"
    data-dimension="xl"
    type="button"
  >
    <span
      class="c1 c2"
    >
      <span>
        Button
      </span>
    </span>
  </button>
</div>
`;
